tests/lib/rules/no-danger.js

```diff
@@ -32,6 +32,26 @@ ruleTester.run('no-danger', rule, {
     { code: '<App />;' },
     { code: '<App dangerouslySetInnerHTML={{ __html: "" }} />;' },
     { code: '<div className="bar"></div>;' },
+    {
+      code: '<div className="bar"></div>;',
+      options: [{ customComponentNames: ['*'] }],
+    },
+    {
+      code: `
+        function App() {
+          return <Title dangerouslySetInnerHTML={{ __html: "<span>hello</span>" }} />;
+        }
+      `,
+      options: [{ customComponentNames: ['Home'] }],
+    },
+    {
+      code: `
+        function App() {
+          return <TextMUI dangerouslySetInnerHTML={{ __html: "<span>hello</span>" }} />;
+        }
+      `,
+      options: [{ customComponentNames: ['MUI*'] }],
```

Comment:
```suggestion
      options: [{ customComponentNames: ['*MUI'] }],
```

Reply:
aha, ok thanks. can you add the same exact test case with the matching pattern, as well?

Reply:
I have added that test case [here](https://github.com/jsx-eslint/eslint-plugin-react/pull/3748/files#diff-4b8ada5e2441fe2af247dda8f99e5c6d99df1b6cbe622f21c8dc6d7cef18ceadR90-R103)

Reply:
the test is good, but i'm suggesting having *identical* cases where the only difference is the glob pattern

Reply:
Added same testcase with opposite pattern in invalid cases.